packages/worker/src/connection-options.ts

```diff
@@ -28,11 +28,18 @@ export interface NativeConnectionOptions {
    * Set statically at connection time, can be replaced later using {@link NativeConnection.setMetadata}.
    */
   metadata?: Record<string, string>;
+
+  /**
+   * API key for Temporal. This becomes the "Authorization" HTTP header with "Bearer " prepended.
+   * This is only set if RPC metadata doesn't already have an "authorization" key.
```

Comment:
> This is only set if RPC metadata doesn't already have an "authorization" key.

On creation of a `NativeConntection`, we should check for apikey+authorization metadata, and throw just like in client connection.